package.xml

```diff
@@ -11,6 +11,7 @@
   <exec_depend>lifecycle_msgs</exec_depend>
   <exec_depend>python3-qt5-bindings</exec_depend>
   <exec_depend>python3-networkx</exec_depend>
+  <exec_depend>python3-scipy</exec_depend>
```

Comment:
There are no direct dependencies in this package on `scipy`, so this should not be required.

Can you be more specific on what failures you are getting?

Reply:
In networkx, they denote scipy as optional. 

https://github.com/networkx/networkx/blob/079c04a4b02ba0f0dd1ab1e05ec5bde62490a95b/pyproject.toml#L68

Reply:
I tried changing the rosdep key for networkx to networkx[extras] but that doesn't work.
```
ERROR: the following packages/stacks could not have their rosdep keys resolved
to system dependencies:
ros_network_viz: Cannot locate rosdep definition for [python3-networkx[extras]]
```
https://stackoverflow.com/questions/6237946/optional-dependencies-in-distutils-pip

Perhaps an alternative rosdep rule needs to be added for `networkx-extras`?

Reply:
Yeah, I see what you mean.  This is happening because we are using the kamada-kawai layout.  In all honest, networkx should really define that as a hard dependency; certain parts of the networkx core won't work without it.

But that is the situation we are in today.  So with that said, I'm happy to take this PR as-is, with a comment added saying why we are adding the `scipy` dependency.

Reply:
> Sounds good. Would you like me to file a rosdep feature request for a generic way to allow packages to support a `python-pkg-name[extras]` specifier? That way this package could depend on `networkx[extras]`?

Feel free to open an issue, though I suspect that this will be tough to do in a package-manager abstract way.

Reply:
> kamada_kawai_layout

Just cleaning up old PR's. I added the comment and rebased. Ready for another pass.